### PR TITLE
Add PyCharm + Docker setup instructions

### PR DESCRIPTION
**Pull request by Samwalton9**

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -75,3 +73,24 @@ All four places referenced there must be updated:
...
+
+## PyCharm setup
+
+This project can be set up via PyCharm using its support for Docker. Wikimedia developers can get free access to PyCharm Professional (required for Docker support) - please contact The Wikipedia Library for instructions.
+
+### Process
+
+1. After installing Docker, ensure it's using Linux Containers, and in settings, enable the 'Expose daemon on tcp://localhost:2375 without TLS' option in settings.
+2. In PyCharm, open the repository folder, and navigate to Project Settings (File > Settings)
+3. Under Build, Execution, Deployment > Docker, click the + symbol and check the TCP socket option is selected.
```
Comment (uyscuti-wiki) [inline docs/developer.md]: and the Engine API URL is `tcp://localhost:2375`

File: docs/developer.md [hunk 2]
```diff
@@ -75,3 +73,24 @@ All four places referenced there must be updated:
...
+4. Navigate to Project > Project Interpreter, click the cog icon, then Add... and select the Docker Compose tab
```
Comment (uyscuti-wiki) [inline docs/developer.md]: cog/gear

File: docs/developer.md [hunk 3]
```diff
@@ -75,3 +73,24 @@ All four places referenced there must be updated:
...
+- Configuration file(s) should be set to both `docker-compose.yml` and `docker-compose.override.yml`
+- Service should be set to `twlight`.
+- Set Python interpreter path to /venv/bin/python
+5. It may take some time for the interpreter to finish adding. If successful, you should see a long list of python packages including Django.
+6. Open the Docker tab in PyCharm's bottom bar. With the Docker entry highlighted, click the icon with three green arrows, then `Create docker-compose deployment...`
+- Compose file(s) should again be set to both `docker-compose.yml` and `docker-compose.override.yml`
+- Check the `--build` option
+- Click Apply, then Run
+7. You should see the three containers - `twlight_docker_db_1`, `twlight_docker_twlight_1`, and `twlight_docker_web_1` start up correctly.
```
Comment (uyscuti-wiki) [inline docs/developer.md]: you should also see the Library Card platform when you visit `localhost` in your browser.

File: docs/developer.md [hunk 4]
```diff
@@ -75,3 +73,24 @@ All four places referenced there must be updated:
...
+8. Whenever you want to start the Docker containers again, you can simply click the green arrows, then 'Docker'.
```
Comment (uyscuti-wiki) [inline docs/developer.md]: Would be worth including two of the potential issues I/we came across and their solutions. Namely, `Cannot start service db:` and `Drive has not been shared`. As an additional tip would be nice to include the process to execute scripts and maybe restart gunicorn.

File: docs/developer.md [hunk 5]
```diff
@@ -2,21 +2,19 @@
...
+These scripts can be run with, for example `docker exec twlight virtualenv_migrate.sh`.
 
 ### Virtual environment
 
 To activate the Python virtual environment for the project, run:
 
-`sudo su www`
-
-`source /var/www/html/TWLight/bin/virtualenv_activate.sh`
+`/app/bin/virtualenv_activate.sh`
```
Comment (jsnshrmn) [inline docs/developer.md]: Is this a thing that still needs to happen? `docker exec twlight bash` should already bring you into an activated environment.

Comment (Samwalton9) [reply docs/developer.md]: Ah, you're right. Removed that section.

**Comment by Samwalton9** [comment]
Done and done :)

**Pull request merged**